### ISSUE TITLE
[Feature Request]: automatically upgrade Nix

### ISSUE THREAD

**Issue by mjgallag** [jetpack-io/devbox#933]

**Comment by Lagoja** [comment]
@mjgallag thank you for reporting! Just to clarify -- you're asking for Devbox to update the version of the `nix` binaries themselves?

**Comment by mjgallag** [comment]
@Lagoja correct, Devbox automatically installed Nix: the package manager v2.13.3 for me but now Nix: the package manager v2.15 is the current version so I want Devbox to also automatically manage Nix: the package manager upgrades. I assume if I take no manual upgrade action I'll eventually be out of date enough to cause trouble. Thanks!

Loving Devbox so far!

**Comment by Lagoja** [comment]
Got it, I think this is a great suggestion. We should roll Nix package manager upgrades into Devbox releases when they line up..

**Comment by bbugh** [comment]
This seems like something you'd want to configure though. Nix is supposed to ensure reproducibility and I don't want the tool I'm using for reproducibility to change itself without asking. 

Maybe this is something that could be added to devbox.json as configuration or devbox.lock, or a warning printed to the console (e.g. like yarn and other package managers).

**Comment by dudicoco** [comment]
@bbugh this also relates to https://github.com/jetify-com/devbox/issues/1371.

Both the devbox version and the nix version should be locked to ensure total reproducibility.
